### PR TITLE
Always get the `NetworkId` when building and balancing

### LINKED ISSUE #213: `NetworkIdMissing` when balancing a tx built via constraints/lookups

### PR DESCRIPTION
**Pull request by ngua**
Closes #213

### CODE REVIEW

**Review by vvtran — APPROVED** [review]
LGTM, some comments.

File: src/Types/ScriptLookups.purs [hunk 1]
```diff
@@ -432,12 +444,7 @@ type ConstraintsConfig (a :: Type) =
 -- We write `ReaderT QueryConfig Aff` below since type synonyms need to be fully
 -- applied.
 type ConstraintsM (a :: Type) (b :: Type) =
```
Comment (vvtran) [inline src/Types/ScriptLookups.purs:446]: I think  `ConstraintsM` comments will need updating.

File: src/BalanceTx.purs [hunk 2]
```diff
@@ -246,7 +245,9 @@ calculateMinFee' = calculateMinFee >>> map (rmap unwrap)
 -- utxo set shouldn't include the collateral which is vital for balancing.
 -- In particular, the transaction inputs must not include the collateral.
 balanceTx :: UnbalancedTx -> QueryM (Either BalanceTxError Transaction)
-balanceTx (UnbalancedTx { transaction: unbalancedTx, utxoIndex }) =
+balanceTx (UnbalancedTx { transaction: unbalancedTx, utxoIndex }) = do
+  networkId <- (unbalancedTx ^. _body <<< _networkId) #
```
[the remark this answers is not in the document]
Comment (ngua) [reply src/BalanceTx.purs:249]: That's true, but when I removed them it was harder to read honestly

[the remark this answers is not in the document]
Comment (vvtran) [reply src/BalanceTx.purs:249]: In that case, feel free to keep it.

**Pull request merged**